client/finality-grandpa/src/finality_proof.rs

```diff
@@ -148,8 +151,82 @@ impl<Block: BlockT> AuthoritySetForFinalityChecker<Block> for Arc<dyn FetchCheck
 	}
 }
 
+/// Justification for a finalized block.
+#[derive(Clone, Serialize, Deserialize)]
+pub struct JustificationNotification<Block: BlockT> {
+	/// Highest finalized block header
+	pub header: Block::Header,
+	/// An encoded justification proving that the given header has been finalized
+	pub justification: Vec<u8>,
+}
+
+type JustificationStream<Block> = TracingUnboundedReceiver<JustificationNotification<Block>>;
```

Comment:
I think all the changes in this file should be moved to a new file (they don't relate to finality proofs at all). Maybe `notification.rs`?